src/ui/controllers/booking/booking.ts

```diff
@@ -5,84 +5,16 @@ import * as learnerRecord from 'lib/learnerrecord'
 import * as notify from 'lib/service/notify'
 import * as template from 'lib/ui/template'
 import * as xapi from 'lib/xapi'
-import * as log4js from 'log4js'
-import * as courseController from './course/index'
 
+import * as courseController from '../course/index'
+
+import * as log4js from 'log4js'
 const logger = log4js.getLogger('controllers/booking')
 
-enum confirmedMessage {
+export enum confirmedMessage {
 	Booked = 'Booked',
 	Cancelled = 'Cancelled',
-}
-
-export async function renderCancelBookingPage(
-	ireq: express.Request,
-	res: express.Response
-) {
-	const req = ireq as extended.CourseRequest
-	const course = req.course
-	const module = req.module!
-	const event = req.event!
-
-	const record = await learnerRecord.getRecord(req.user, course, module, event)
-
-	if (!record) {
-		logger.warn(
```

Comment:
This is repeated three times, better to abstract to a function rather than repeat the same thing three times. If you rebase you'll find this has been done in booking though!